src/CognitiveServices/SpeechSynthesis.ts

```diff
@@ -0,0 +1,517 @@
+import { Speech, Action } from '../SpeechModule'
+import { konsole } from '../Chat';
+
+export interface ICognitiveServicesSpeechSynthesisProperties {
+    subscriptionKey: string,
+    gender?: SynthesisGender,
+    voiceName?: string
+}
+
+export enum SynthesisGender { Male, Female };
+
+interface SpeakRequest {
+    isReadyToPlay: boolean,
+    data: ArrayBuffer,
+    text: string,
+    locale: string,
+    onSpeakingStarted: Action,
+    onSpeakingFinished: Action,
+}
+
+interface HttpHeader {
+    name: string,
+    value: string
+}
+
+export class SpeechSynthesizer implements Speech.ISpeechSynthesizer {
+    private _requestQueue: SpeakRequest[] = null;
+    private _isPlaying: boolean = false;
+    private _audioElement: AudioContext;
+    private _helper: CognitiveServicesHelper;
+    private _properties: ICognitiveServicesSpeechSynthesisProperties;
+
+    constructor(properties: ICognitiveServicesSpeechSynthesisProperties) {
+        this._helper = new CognitiveServicesHelper(properties.subscriptionKey);
+        this._properties = properties;
+        this._audioElement = new AudioContext();
+        this._requestQueue = new Array();
+    }
+
+    speak(text: string, lang: string, onSpeakingStarted: Action = null, onSpeakingFinished: Action = null): void {
+        this._requestQueue.push(
+            {
+                isReadyToPlay: false,
+                data: null,
+                text: text,
+                locale: lang,
+                onSpeakingStarted: onSpeakingStarted,
+                onSpeakingFinished: onSpeakingFinished
+            }
+        );
+
+        this.getSpeechData().then(() => {
+            this.playAudio();
+        });
+
+    }
+
+    stopSpeaking(): void {
+        if (this._isPlaying) {
+            this._requestQueue = [];
+            this._isPlaying = false;
+            if(this._audioElement.state !== "closed"){
+                this._audioElement.close();
+            }
+        }
+    }
+
+    private playAudio() {
+        if (this._requestQueue.length == 0) {
+            return;
+        }
+
+        let top = this._requestQueue[0];
```

Comment:
Please use `const` whenever a variable won't be reset, such as with `top` here. This is consistent with the rest of WebChat.